src/openrct2/core/String.cpp

```diff
@@ -557,8 +561,46 @@ namespace String
 
         return dst;
 #else
-        STUB();
-        return std::string(src);
+        const char* codepage;
+        switch (srcCodePage)
+        {
+        case CODE_PAGE::CP_932:
+            codepage = "windows-932";
+            break;
+
+        case CODE_PAGE::CP_936:
+            codepage = "GB2312";
+            break;
+
+        case CODE_PAGE::CP_949:
+            codepage = "windows-949";
+            break;
+
+        case CODE_PAGE::CP_950:
+            codepage = "big5";
+            break;
+
+        default:
+            throw std::runtime_error("Unsupported code page: " + std::to_string(srcCodePage));
+        }
+
+        icu::UnicodeString convertString(src.data(), codepage);
+        std::string result;
+
+        if (dstCodePage == CODE_PAGE::CP_UTF8)
```

Comment:
Note: this isn't converting from UTF-8 to any other code page yet.

Reply:
Is it not possible for you to feed the same code page IDs to ICU?

Reply:
Are there any functions in libicu to convert from number to name?

Reply:
Not as far as I'm aware.  I think the best option will be something like `sprintf(s, "windows-%s", codepage)`, but I'm not sure that works for everything, so I'll have to check that.

Reply:
[ucnv_getStandard](https://ssl.icu-project.org/apiref/icu4c/ucnv_8h.html#a96c90729859689a2ae7c29f3f7f1306a) seems to be the function we're looking for.

Reply:
Hmm, are you sure?  I was under the impression that that function took an ICU-internal index.